workflow/util/util.go

```diff
@@ -731,6 +734,30 @@ func RetryWorkflow(kubeClient kubernetes.Interface, hydrator hydrator.Interface,
 		}
 	}
 
+	if len(deletedNodes) > 0 {
+		for _, node := range newWF.Status.Nodes {
+			if len(node.Children) > 0 {
```

Comment:
minor: I don't think you need this guard - in `len(children) == 0`, then `newChildren` will be `nil`, so `node.Children` will become/stay nil

Reply:
Yup! Good point. Done.